### PR TITLE
Refactor CMakeLists into distinct directories

### PR DESCRIPTION
**Pull request by michael-loki**

### CODE REVIEW

**Comment by michael-loki** [comment]
this is ready to merge. I've tested on all platforms I have access to (but not windows).

File: test/dht/test_llarp_dht_key.cpp [hunk 1]
```diff
@@ -30,15 +30,24 @@ TEST_P(DHT, constructor)
...
+static constexpr Array fullArray{
+    {0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+     0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+     0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff}};
 
-static constexpr Array seqArray {{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F,0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F}};
+static constexpr Array seqArray{
+    {0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A,
+     0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15,
+     0x16, 0x17, 0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F}};
```
Comment (neuroscr) [inline test/dht/test_llarp_dht_key.cpp]: stop doing this, macos needs those double braces and I'm sure clang-format is making me revert this.

File: test/dht/test_llarp_dht_key.cpp [hunk 2]
```diff
@@ -30,15 +30,24 @@ TEST_P(DHT, constructor)
...
 
 static const Array data[] = {emptyArray, fullArray, seqArray};
 
-INSTANTIATE_TEST_CASE_P(TestDhtKey, DHT, ::testing::ValuesIn(data));
+INSTANTIATE_TEST_CASE_P(TestDhtKey, DHT, ::testing::ValuesIn(data), );
```
Comment (neuroscr) [inline test/dht/test_llarp_dht_key.cpp]: hanging comma right?

File: vendor/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  cppbackport-master/lib/fs/path.cpp
+  cppbackport-master/lib/fs/remove.cpp
+  cppbackport-master/lib/fs/diriter.cpp
+  cppbackport-master/lib/fs/copyfile.cpp
+  cppbackport-master/lib/fs/absolute.cpp
+  cppbackport-master/lib/fs/direntry.cpp
+)
+
+add_library(cppbackport STATIC ${CXX_COMPAT_SRC})
+target_include_directories(cppbackport PUBLIC cppbackport-master/lib)
```
Comment (neuroscr) [inline vendor/CMakeLists.txt]: should tuntap go here too?

File: test/dht/test_llarp_dht_key.cpp (hunk 1 above)
Comment (michael-loki) [reply test/dht/test_llarp_dht_key.cpp]: This was `clang-format` doing this for me? I can't reproduce on my version of macOS, what version of AppleClang are you on?

File: vendor/CMakeLists.txt (hunk 3 above)
Comment (michael-loki) [reply vendor/CMakeLists.txt]: `tuntap` is a different kettle of fish, its pulled into the platform lib.

File: test/dht/test_llarp_dht_key.cpp (hunk 1 above)
Comment (neuroscr) [reply test/dht/test_llarp_dht_key.cpp]: Apple LLVM version 8.1.0 (clang-802.0.42)

Interesting that our clang-formats are doing different things:

$ clang-format --version
clang-format version 6.0.0 (tags/RELEASE_600/final)

Comment (michael-loki) [reply test/dht/test_llarp_dht_key.cpp]:
```
$ clang-format --version
clang-format version 8.0.0 (tags/google/stable/2018-12-18)
```

**Comment by despair86** [comment]
yeah this doesn't fly over here

inb4 lokinet is finally limited to linux and mac by v1

**Comment by majestrate** [comment]
On Mon, Feb 11, 2019 at 06:19:19PM -0800, Rick V wrote:
> yeah this doesn't fly over here

what part of it doesn't work out for you specifically?

**Comment by michael-loki** [comment]
This is ready to merge @majestrate - I had a build work + pass tests on windows, mac and linux.

File: test/win32/test.rc [hunk 4]
```diff
@@ -6,7 +6,7 @@
 //
 // Microsoft Visual C++ generated resource script.
 //
-#include "resource.h"
+#include <win32/resource.h>
 #include <winresrc.h>
```
Comment (despair86) [inline test/win32/test.rc]: not strictly necessary, but then i assume that App Studio ended up rewriting some part of the code later on (i.e. stuff like `LANGUAGE` directive

File: test/dht/test_llarp_dht_key.cpp (hunk 1 above)
Comment (neuroscr) [reply test/dht/test_llarp_dht_key.cpp]: Either way we need to update the clang-format file for the newer versions, so it stops

**Pull request merged**